File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/Gender/GenderColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB.Columns;

[ColumnsScript("CrmDB.Gender")]
[BasedOnRow(typeof(GenderRow), CheckNames = true)]
public class GenderColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int GenderId { get; set; }
    [EditLink]
    public string Name { get; set; }
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/GenderColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { GenderRow } from "./GenderRow";

export interface GenderColumns {
    GenderId: Column<GenderRow>;
    Name: Column<GenderRow>;
}

export class GenderColumns extends ColumnsBase<GenderRow> {
    static readonly columnsKey = 'CrmDB.Gender';
    static readonly Fields = fieldsProxy<GenderColumns>();
}